### ISSUE TITLE
Turn the cache key in CircleCI config into a variable

### ISSUE THREAD

**Issue by chinedufn** [chinedufn/percy#88]
It's repeated several times across the file. We can do something like this:

https://gist.github.com/bschwind/cb11c4c46710732b034ddcd99f3ab0d8#file-config-yml-L1-L2

In general we should abstract some of the duplicated bits of the config for easier readability/maintainability

**Comment by dbrgn** [comment]
Yep. The main issue is that YAML can't concatenate lists, so you can't extract shared steps into references...

CircleCI have added a custom layer on top of YAML: https://circleci.com/docs/2.0/reusing-config/ That should help, but it adds more complexity where I'm not sure whether it's really worth it. There's less duplication but the config is harder to follow when reading it.

**Comment by chinedufn** [comment]
Ahhh - ok I'll take a look at that link - but are you saying that we can't really do much better?

**Comment by chinedufn** [comment]
Also we can still turn the key into a variable - yeah? Sorry I'm a circle config noob so just verifying!

**Comment by dbrgn** [comment]
We can, but note that every job needs its own cache key. That means that we'd save 2 lines per job, but at the same time we'd have to add a reference which is also 2 lines and adds indirection.

You could argue that it reduces the chance of a cache key mismatch (when saving and restoring), but it also makes the config file more complex and harder to read.

So I don't think we can do much better unfortunately... In my opinion, sometimes duplication is OK if it improves readability :slightly_smiling_face:

**Comment by chinedufn** [comment]
Gotcha - I didn't even know a lot of the points that you brought up and thought that there was a bigger opportunity here ... - sounds good!!!

**Comment by chinedufn** [comment]
Whoops re-opening as I still think that having the cache keys at the top would make it much less typo prone..

But I see that we can't do very much here
